packages/node-opentelemetry/src/instrumentations.ts

```diff
@@ -0,0 +1,251 @@
+import { ClientRequest, IncomingMessage, ServerResponse } from 'http';
+
+import * as shimmer from 'shimmer';
+import _ from 'lodash';
+import { Span as ISpan } from '@opentelemetry/api';
+
+import {
+  Logger,
+  LoggerOptions,
+  parseWinstonLog,
+} from '@hyperdx/node-logger/build/src/logger';
+
+import hdx from './debug';
+
+export const _parseConsoleArgs = (args: any[]) => {
+  const stringifiedArgs = [];
+  let firstJson;
+  for (const arg of args) {
+    if (_.isObject(arg)) {
+      if (firstJson == null && _.isPlainObject(arg)) {
+        firstJson = arg;
+      }
+      try {
+        const stringifiedArg = JSON.stringify(arg);
+        if (stringifiedArg != null) {
+          stringifiedArgs.push(stringifiedArg);
+        }
+      } catch (e) {
+        // ignore
+      }
+    } else {
+      stringifiedArgs.push(arg);
+    }
+  }
+
+  return firstJson
+    ? {
+        ...firstJson,
+        // FIXME: we probably don't want to override 'message' field in firstJson
+        ...(args.length > 1 && { message: stringifiedArgs.join(' ') }),
+      }
+    : stringifiedArgs.join(' ');
+};
+
+const ALL_HTTP_REQUEST_HEADERS = [
```

Comment:
I'm concerned this only really captures an arbitrary(?) subset of headers, wondering if we should just bring the [otel header capture implementation](https://github.com/open-telemetry/opentelemetry-js/blob/902229afd89a2cf59b120b75892a56bdab5ff039/experimental/packages/opentelemetry-instrumentation-http/src/http.ts#L345) directly into our req/res hooks if possible?

If not possible, we should probably at least document this default behavior, and give a configuration option so anyone can pass their own custom headers.

Reply:
Pushed the update. Now all headers should be captured